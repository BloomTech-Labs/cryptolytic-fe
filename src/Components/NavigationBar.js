import React from "react";
import { NavLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import logo from "../images/logo.svg";
import home from "../images/home-logo.svg";
import arbitrage from "../images/arbitrage-logo.svg";
import trading from "../images/line-chart-logo.svg";
import candlestick from "../images/candlestick-icon2.png";
import api from "../images/cloud-upload-logo.svg";
import data from "../images/data-logo.svg";
import Link from "@material-ui/core/Link";

import AlertDialog from "./AlertDialog";

const useStyles = makeStyles(theme => ({
	NavBar: {
		width: "17vw",
		height: "100vh",
		backgroundColor: "black",
		boxShadow: "1px 0px 0px",
		borderRight: "1px solid #FFFFFF",
		position: "fixed",
		top: 0,
		left: 0,
		zIndex: 100,
		"@media(max-width: 1300px)": {
			width: "100vw",
			height: "10vh",
			position: "static"
		}
	},
	NavBarLogo: {
		width: "100%",
		height: "100px",
		display: "flex",
		justifyContent: "center",
		alignItems: "center",
		"@media(max-width: 1300px)": {
			display: "none"
		}
	},
	NavBarMenu: {
		paddingTop: "3vh",
		width: "100%",
		display: "flex",
		alignItems: "center",
		flexDirection: "column",
		color: "#FFFFFF",
		marginLeft: 15,
		"@media(max-width: 1300px)": {
			flexDirection: "row",
			padding: 0,
			margin: 0,
			height: "10vh"
		}
	},
	navBarMenuText: {
		width: "100%",
		textAlign: "left",
		fontWeight: 600,
		fontSize: "1.2rem",
		lineHeight: "33px",
		letterSpacing: "0.03em",
		display: "flex",
		textDecoration: "none",
		color: "#fff"
	},
	navBarMenuTextDisabled: {
		width: "100%",
		textAlign: "left",
		fontWeight: 600,
		fontSize: "1.2rem",
		lineHeight: "33px",
		letterSpacing: "0.03em",
		display: "flex",
		textDecoration: "none",
		color: "gray"
	},
	NavBarMenuContainer: {
		"@media(max-width: 1300px)": {
			display: "flex",
			width: "85%"
		},
		"&:hover": {
			backgroundColor: "rgba(35, 32, 44, 0.9)",
			width: "auto",
			marginLeft: 20,
			position: "relative"
		}
	},
	navBarMenuTextCursorActive: {
		backgroundColor: "#4EB9FF",
		width: 20,
		marginLeft: "-18vw",
		"@media(max-width: 1300px)": {
			width: 0,
			marginLeft: "0vw"
		}
	},
	navBarMenuButton: {
		visibility: "visible",
		display: "flex",
		marginBottom: "10px",
		alignItems: "center",
		width: "100%",
		height: "100%",
		position: "relative",
		transition: "0.2s ease",
		"@media(max-width: 1300px)": {
			marginBottom: "20px",
			"&:hover": {
				boxShadow: "1px 10px #4EB9FF"
			}
		}
	},
	navBarMenuIcon: {
		margin: "0px 10px",
		marginLeft: 50
	}
}));

function NavigationBar() {
	const classes = useStyles();
	return (
		<div className={classes.NavBar}>
			<div className={classes.NavMenuContainer}>
				<div className={classes.NavBarLogo}>
					<Link href='/'>
						<img src={logo} className={classes.Logo} alt='logo' />
					</Link>
				</div>
				<div className={classes.NavBarMenu}>
					{[
						{ name: "Home", logo: home, link: "home", status: "active" },
						{
							name: "Arbitrage",
							logo: arbitrage,
							link: "arbitrage-dashboard",
							status: "active"
						},
						{
							name: "Trading",
							logo: trading,
							link: "trading-dashboard",
							status: "active"
						},
						{
							name: "Charts",
							logo: candlestick,
							link: "exchange-charts",
							status: "active"
						},
						{ name: "API", logo: api, link: "api", status: "inactive" },
						{ name: "Data", logo: data, link: "data", status: "inactive" }
					].map(e =>
						e.status === "inactive" ? (
							<AlertDialog
								button={
									<NavLink
										activeClassName={classes.navBarMenuTextCursor}
										to={e.link}
										className={classes.navBarMenuTextDisabled}
									>
										<div className={classes.navBarMenuButton}>
											<img
												className={classes.navBarMenuIcon}
												style={{ margin: "0px 10px", marginLeft: 50 }}
												src={e.logo}
												alt={e.name}
											/>
											{e.name}
										</div>
									</NavLink>
								}
								title={"Coming Soon"}
								message={`We're still working on features included in the ${e.name} page. Come back soon!`}
							/>
						) : (
								<div style={{ all: "inherit" }}>
									<NavLink
										activeClassName={classes.navBarMenuTextCursorActive}
										to={e.link}
										className={classes.navBarMenuText}
									>
										<div className={classes.navBarMenuButton}>
											<img
												className={classes.navBarMenuIcon}
												src={e.logo}
												alt={e.name}
											/>
											{e.name}
										</div>
									</NavLink>
								</div>
							)
					)}
				</div>
			</div>
		</div>
	);
}

export default NavigationBar;
