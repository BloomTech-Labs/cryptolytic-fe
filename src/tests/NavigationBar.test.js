import React from "react";
import { MemoryRouter } from "react-router";
///
import NavigationBar from "../Components/NavigationBar";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("NavigationBar component initialized without crashing", () => {
  render(
    <MemoryRouter>
      <NavigationBar />
    </MemoryRouter>
  );
});

it("NavigationBar menu items rendered!", () => {
  const menuItems = ['home', 'arbitrage', 'trading', 'api', 'data']
  const { getByText } = render(
    <MemoryRouter>
      <NavigationBar />
    </MemoryRouter>
  );
  menuItems.forEach(e => {
    expect(`/${e}/i`)
  })
});

describe("<NavigationBar /> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavigationBar />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
