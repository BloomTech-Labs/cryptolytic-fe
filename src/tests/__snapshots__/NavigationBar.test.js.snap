// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> matches verified SnapShot should match snapshot 1`] = `
<div
  className="makeStyles-NavBar-121"
>
  <div>
    <div
      className="makeStyles-NavBarLogo-122"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <img
          src="logo.svg"
        />
      </a>
    </div>
    <div
      className="makeStyles-NavBarMenu-123"
    >
      <div
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuText-124"
          href="/home"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="home-logo.svg"
            />
            Home
          </div>
        </a>
      </div>
      <div
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuText-124"
          href="/arbitrage-dashboard"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="arbitrage-logo.svg"
            />
            Arbitrage
          </div>
        </a>
      </div>
      <div
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuText-124"
          href="/trading-dashboard"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="line-chart-logo.svg"
            />
            Trading
          </div>
        </a>
      </div>
      <div
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuText-124"
          href="/exchange-charts"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="candlestick-icon2.png"
            />
            Charts
          </div>
        </a>
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuTextDisabled-125"
          href="/api"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="cloud-upload-logo.svg"
              style={
                Object {
                  "margin": "0px 10px",
                  "marginLeft": 50,
                }
              }
            />
            API
          </div>
        </a>
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "all": "inherit",
          }
        }
      >
        <a
          aria-current={null}
          className="makeStyles-navBarMenuTextDisabled-125"
          href="/data"
          onClick={[Function]}
        >
          <div
            className="makeStyles-navBarMenuButton-128"
          >
            <img
              className="makeStyles-navBarMenuIcon-129"
              src="data-logo.svg"
              style={
                Object {
                  "margin": "0px 10px",
                  "marginLeft": 50,
                }
              }
            />
            Data
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
